ssl/statem/statem_srvr.c

```diff
@@ -355,6 +372,26 @@ static int send_server_key_exchange(SSL_CONNECTION *s)
     return 0;
 }
 
+/*
+ * Should we send a CompressedCertificate message?
+ *
+ * Valid return values are:
+ *   1: Yes
+ *   0: No
+ */
+static int send_compressed_certificate(SSL_CONNECTION *sc)
+{
+#ifndef OPENSSL_NO_COMP_ALG    
+    int alg = sc->ext.compress_certificate_from_peer;
```

Comment:
It would be better to have only selected a certificate algorithm that we know we have a compressed certificate for. E.g. say we have a compressed certificate for zstd but not brotli, we could end up deciding that brotli is the most preferred algorithm, even though we don't have a compressed cert for it. Then subsequently we will end up sending an uncompressed cert - even if both endpoints can handle zstd.

Reply:
Agreed. This is basically what I said on one of your earlier comments. This is where the calculation of the algorithm should be done. The extension processing code should just save the peer's algorithms.